lightningd/offer.c

```diff
@@ -14,17 +14,40 @@ static void json_populate_offer(struct json_stream *response,
 				const struct sha256 *offer_id,
 				const char *b12,
 				const struct json_escape *label,
+				const char *description,
 				enum offer_status status)
 {
 	json_add_sha256(response, "offer_id", offer_id);
 	json_add_bool(response, "active", offer_status_active(status));
 	json_add_bool(response, "single_use", offer_status_single(status));
 	json_add_string(response, "bolt12", b12);
+	if (description)
```

Comment:
Adding `description` here causes it to appear in the `offer` RPC response as well, which results in a msggen error because that field already exists in the `offer` request. We can rather add `description` separately in `listoffers`, `disableoffer` and `enableoffer` responses.